### PR TITLE
MAILBOX-371 Test out of order registrations

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java [hunk 1]
```diff
@@ -163,6 +163,18 @@ default void unregisterShouldRemoveDoubleRegisteredListener() throws Exception {
                 .event(any());
         }
 
+        @Test
```
Comment (hoangdat) [inline mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: base on IMAP session consistency, please add test in the case Event be dispatched from other event bus.
```java
       default void registerShouldNotDispatchPastEvents() throws Exception {
            MailboxListener listener = newListener();

             eventBus2().dispatch(EVENT, ImmutableSet.of(KEY_1)).block();

             eventBus().register(listener, KEY_1);

             verify(listener, after(FIVE_HUNDRED_MS).never())
                .event(any());
        }
```

Comment (hoangdat) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: I think I will do it in my PR

Comment (chibenwa) [reply mailbox/api/src/test/java/org/apache/james/mailbox/events/KeyContract.java]: Will add it here

**Comment by chibenwa** [comment]
```
[44368a12948e626480ac830dbde40d97a53fa0fd] [ERROR]   FixingGhostMailboxTest.webadminCanMergeTwoMailboxes:224 1 expectation failed.
[44368a12948e626480ac830dbde40d97a53fa0fd] JSON path [0][1].messageIds doesn't match.
[44368a12948e626480ac830dbde40d97a53fa0fd] Expected: a collection with size <2>
[44368a12948e626480ac830dbde40d97a53fa0fd]   Actual: [74ca27b0-1b06-11e9-a4c1-392d534d1d8b]
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**